src/drm.h

```diff
@@ -211,10 +211,6 @@ class Plane : public Object {
 
 class FrameBuffer : public Object {
  public:
-  struct Plane {
```

Comment:
This has to be re-added, breaks the build, it's because cppcheck find's it hard to deduce auto, sometimes humans find it hard to deduce auto (like me :smile: ) so I have mixed feelings about it also. Would prefer to specify the type explicitly to satisfy the tool. If you change:

`for (const auto& plane : planes_) {`

To it's actual type, the warning should go away, it's probably something like `const std::pair<int, Plane>& plane` because planes_ is a `std::map<int, Plane>`. There is a way the compiler can tell you below:

https://stackoverflow.com/questions/17818009/how-can-i-determine-the-actual-type-of-an-auto-variable

Reply:
It's a false positive from cppcheck in other words

Reply:
Ack removed the header file checks, see next pushes

Reply:
So i get the same error with this:
```
for (const std::pair<const int, Plane>& plane : planes_) {
```
I do not understand the logic,  cause the error

class FrameBuffer : public Object {
 public:
  struct Plane {
    uint32_t handle;
  };
  ~FrameBuffer();
  
  But the referred line and the usage give a hint it is not the same reference:
  ```
    for (const std::pair<const int, Plane>& plane : planes_) {
         struct drm_gem_close gem_close;
        gem_close.handle = **plane.second.handle**;
  ```

Reply:
I'm gonna take a look asap

Reply:
Thanks for trying @Yarboa

Reply:
This header can remain as is unless there are other non-unusedStructMember issues to fix

Reply:
No this is the only one as per now, ok

Reply:
Returned suppression unusedStructMember